### ISSUE TITLE
Add ability to pass Element itself to targetModal, not only 'id' of element

### ISSUE THREAD

**Issue by 4esnog** [ghosh/Micromodal#65]
Hello!

I am using shadow DOM, and if my modal is hidden inside it, `document.getElementById('id-of-shadowed-element')` can't select my modal.

I think, it would be handy to have the ability to pass not only id of element, but the element itself in `options.targetModal`.

Example:
```
const el = document.querySelector('my-component');
MicroModal.show(el);
```

**Comment by crtl** [comment]
Is this still on the radar?

**Comment by Raidri** [comment]
I push this because I need that feature too for my web components. Is there maybe any new informations?

**Comment by ghosh** [comment]
In an effort to keep this library lightweight and simple, have decided not to implement this feature. If there is a rising demand for this, am open to consider this again.

Thanks for your patience and understanding guys.

**Comment by pbek** [comment]
Being able to pass an element as modal would also make this library work inside [web-components](https://en.wikipedia.org/wiki/Web_Components).

**Comment by pbek** [comment]
@ghosh, I can confirm that changing this one line (https://github.com/ghosh/Micromodal/pull/351/files) allows the library to be used in web components.

![image](https://user-images.githubusercontent.com/1798101/86778009-2810ca00-c05a-11ea-8bbf-20e37cba9e15.png)

**Comment by pbek** [comment]
## Local

![image](https://user-images.githubusercontent.com/1798101/86913108-6a9bda80-c11e-11ea-98ae-0810b246204e.png)

## Nextcloud

![image](https://user-images.githubusercontent.com/1798101/86913130-75566f80-c11e-11ea-9caa-a8d36da117c6.png)

**Comment by gstvribs** [comment]
any updates on this issue?

**Comment by obbap1** [comment]
please any updates on this ?

**Comment by benCoomes** [comment]
Another bump, if this library doesn't support shadow dom then I can't use it for the project I am working on. Which is a shame, because it is forward-looking with no dependencies and es6 module support.

**Comment by pbek** [comment]
Just as a reminder, I have my PR https://github.com/ghosh/Micromodal/pull/351 running in production since July 2020 without any issues in our web components.

**Comment by pbek** [comment]
Any news or considerations on this, @ghosh?

**Comment by dkniffin** [comment]
I've taken over maintenance of micromodal, and I just merged PR #351. I'll be pushing it out in v0.6.0

**Comment by dkniffin** [comment]
v0.6.0 is published! 🚀  Let me know if anyone has any issues with it.
